### DISCUSSION TITLE
DietPi-Software | Shairport Sync: Update to v4.1 with AirPlay 2

### DISCUSSION THREAD

**Discussion by matt-from-uk** [MichaIng/DietPi#5688] [Software Request] [answered]
Hi, would it be possible to add the shareport-sync development branch as an option to install please? 
https://github.com/mikebrady/shairport-sync/tree/development 

Shareport-sync is already an option but the development branch includes a number of notable features missing in the master branch like Airplay 2 support that is of benefit when using DietPi to run a streamer or amp.

I suggest that both are made available from the software install menu so users can choose which branch they prefer.

Many thanks

**Comment by Joulinar** [comment]
Usually, we don't offer development branch on our catalogue. If someone needs this specific branch, he will need to install it on his own. But maybe @MichaIng has a different view.

**Comment by MichaIng** [comment, marked as answer]
I'll update our package to latest release v3.3.9 soon, but maintaining a development package is out of scope. However, I'll upload a build script for GitHub actions, which should be easy to adjust for building own packages from newer branches/commits.

**Reply by matt-from-uk** [reply]
Thanks - unfortunately, that sounds beyond my abilities, but I will try.
Thank you for taking the time to reply

**Comment by MichaIng** [comment]
Linking related request from our forum: https://dietpi.com/forum/t/req-devel-branch-of-shairport-sync-for-airplay-2/11233
However, I'll not add this until there has been a stable release of v4.1: https://github.com/mikebrady/shairport-sync/releases
But builds have been moved to GitHub, making updates and changes to the package easier: https://github.com/MichaIng/DietPi/pull/5804

**Comment by MichaIng** [comment]
Shairport Sync v4.1 with AirPlay 2 support has been released and packages are ready for testing: https://dietpi.com/downloads/binaries/bullseye/testing/
```sh
cd /tmp
curl "https://dietpi.com/downloads/binaries/$G_DISTRO_NAME/shairport-sync-airplay2_$G_HW_ARCH_NAME.deb" -o package.deb
apt install ./package.deb
rm package.deb
systemctl restart shairport-sync
```

**Reply by MichaIng** [reply]
@matt-from-uk would be great if you find time to test the package with AirPlay 2 support. They install fine and services run well, but I have no Apple infrastructure to test actual connectivity and playback.

**Comment by matt-from-uk** [comment]
Hey @MichaIng sorry for the delay.

I don't have any rpi in action at the moment so set up on spare device. All looks good, hostname is shown up on the airplay option to stream from iphone as expected. Selecting the dietpi rpi also allows to then group with other airplay2 devices so I think it is a success. The onlything I couldn't check was audio output as no speakers to hand, but I can at least confirm the ios device airplay2 interaction 'looks' like it is working well even if I can't hear whether it is at present.

Cheers for this.

**Reply by MichaIng** [reply]
Thanks for testing. Yes I think it works fine. Here are some other test results: https://dietpi.com/forum/t/req-devel-branch-of-shairport-sync-for-airplay-2/11233/16?u=michaing
